### PR TITLE
[SIM_FLUTTER-11] [FE] Create User Sign-up Page

### PR DESCRIPTION
**Pull request by jescabillas**
**Issue Link**
- [BackLog](https://framgiaph.backlog.com/view/SIM_FLUTTER-11)
- [Figma](https://www.figma.com/file/uzAwGmOCLvGbReXshnT3UN/SAVINGS-APP?type=design&node-id=484-207&t=jem9jubSN8OujJHz-0)

**Definition of done**
- [x] Create page for sign up step 1
- [x] Create page for sign up step 2
- [x] Follow the figma design
- [x] User can navigate from Log In page to Sign Up page and vice versa using the link text under the submit button

**Note**
- No validation and data yet, this will be applied in integration task
- Added button in the home page to Go to Sign Up page for the mean time

**Attachments**
- Step 1
image
image

- Step 2
image
image

https://github.com/framgia/sph-flutter/assets/114911074/e6fdfe72-aeaf-4bec-b68f-6362047246fe

### CODE REVIEW

**Comment by Dbulawan13** [comment]
Comment:

- [x] Please add the logo and adjust the position of the "Sign up" text slightly lower, based on the design in Figma. (Please apply in both pages)
image

Figma:
image
https://www.figma.com/file/uzAwGmOCLvGbReXshnT3UN/SAVINGS-APP?type=design&node-id=484-207&t=tfLDGtzmPMTOm1OI-0

- [x] Please add the pre-existing email field validation. You can refer to other pull requests that have implemented email field validations.
image

Reference:
https://github.com/framgia/sph-flutter/pull/6#pullrequestreview-1480563799
image

Comments: 2

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #11 

Working:
- Able to create user sign-up page
- Comments have been fixed
- Able to redirect login page to signup page


https://github.com/framgia/sph-flutter/assets/86758474/52062c8b-1454-4d7e-916c-6289b30d2d21

Testing Hours: 1
Comments: 3
Iterations: 2

File: frontend/lib/src/features/sign_up/sign_up_flow.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import 'package:flow_builder/flow_builder.dart';
+
+import 'package:frontend/src/features/sign_up/sign_up_step_one.dart';
+import 'package:frontend/src/features/sign_up/sign_up_step_two.dart';
+
+/*
+  The page where user can see their profile informations.
+*/
+
+class User {
```
Comment (RDLegs) [inline frontend/lib/src/features/sign_up/sign_up_flow.dart]: Move to lib/src/models as `User.dart` and use quicktype.io as reference

Also, this seems to be an implementation specific to FlowBuilder. Would it possible if we create a base class in models that is similar to previous implementations and then create another class extending that for this implementation specifically?

i.e. UserFlowState extends User {}

File: frontend/lib/src/features/sign_up/sign_up_flow.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      username: username ?? this.username,
+      email: email ?? this.email,
+      password: password ?? this.password,
+      passwordConfirmation: passwordConfirmation ?? this.passwordConfirmation,
+      page: page ?? this.page,
+    );
+  }
+}
+
+class SignUpFlow extends StatelessWidget {
```
Comment (RDLegs) [inline frontend/lib/src/features/sign_up/sign_up_flow.dart:65]: Add widget description

File: frontend/lib/src/features/sign_up/sign_up_step_one.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import 'package:get/get.dart';
+
+import 'package:frontend/src/features/sign_up/sign_up_flow.dart';
+import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:frontend/src/components/button.dart';
+import 'package:frontend/src/features/login/login_page.dart';
+import 'package:frontend/src/components/input/date_picker_field.dart';
+
+class SignUpStepOne extends StatelessWidget {
```
Comment (RDLegs) [inline frontend/lib/src/features/sign_up/sign_up_step_one.dart:17]: Add widget description

File: frontend/lib/src/features/sign_up/sign_up_step_two.dart [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:frontend/src/components/auth/auth_header.dart';
+import 'package:get/get.dart';
+
+import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:frontend/src/components/button.dart';
+import 'package:frontend/src/features/login/login_page.dart';
+
+class SignUpStepTwo extends StatelessWidget {
```
Comment (RDLegs) [inline frontend/lib/src/features/sign_up/sign_up_step_two.dart:15]: Add Widget description

**Review by RDLegs — COMMENTED** [review]
Please resolve some of these issues

File: frontend/lib/src/features/sign_up/sign_up_flow.dart (hunk 2 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/sign_up_flow.dart:65]: Added widget description

File: frontend/lib/src/features/sign_up/sign_up_step_one.dart (hunk 3 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/sign_up_step_one.dart:17]: Added widget description

File: frontend/lib/src/features/sign_up/sign_up_flow.dart (hunk 1 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/sign_up_flow.dart]: I've adjusted the class naming please recheck, this class is the state form data for FlowBuilder.

**Review by RDLegs — DISMISSED** [review]
LGTM!

We can just finalize the classes during implemnetation

File: frontend/lib/src/features/sign_up/components/breadcrumb.dart [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/material.dart';
+import 'package:frontend/src/features/sign_up/sign_up_flow.dart';
+import 'package:flow_builder/flow_builder.dart';
```
Comment (rdelrosariojr) [inline frontend/lib/src/features/sign_up/components/breadcrumb.dart:2]: arrange imports

```suggestion
import 'package:flutter/material.dart';
import 'package:flow_builder/flow_builder.dart';

import 'package:frontend/src/features/sign_up/sign_up_flow.dart';
```

File: frontend/lib/src/features/sign_up/sign_up_step_one.dart [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+import 'package:flow_builder/flow_builder.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:get/get.dart';
+
+import 'package:frontend/src/features/sign_up/sign_up_flow.dart';
+import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:frontend/src/components/button.dart';
+import 'package:frontend/src/features/login/login_page.dart';
+import 'package:frontend/src/components/input/date_picker_field.dart';
```
Comment (rdelrosariojr) [inline frontend/lib/src/features/sign_up/sign_up_step_one.dart:11]: Also here
```suggestion
import 'package:flutter/material.dart';
import 'package:flow_builder/flow_builder.dart';
import 'package:get/get.dart';

import 'package:flutter_form_builder/flutter_form_builder.dart';
import 'package:frontend/src/components/auth/auth_header.dart';
import 'package:frontend/src/features/sign_up/sign_up_flow.dart';
import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
import 'package:frontend/src/components/input/input_field.dart';
import 'package:frontend/src/components/button.dart';
import 'package:frontend/src/features/login/login_page.dart';
import 'package:frontend/src/components/input/date_picker_field.dart';
```

File: frontend/lib/src/features/sign_up/sign_up_step_two.dart [hunk 7]
```diff
@@ -0,0 +1,135 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:frontend/src/components/auth/auth_header.dart';
+import 'package:get/get.dart';
+
+import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:frontend/src/components/button.dart';
+import 'package:frontend/src/features/login/login_page.dart';
```
Comment (rdelrosariojr) [inline frontend/lib/src/features/sign_up/sign_up_step_two.dart:9]:
```suggestion
import 'package:flutter/material.dart';
import 'package:flutter_form_builder/flutter_form_builder.dart';
import 'package:form_builder_validators/form_builder_validators.dart';
import 'package:get/get.dart';

import 'package:frontend/src/components/auth/auth_header.dart';
import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
import 'package:frontend/src/components/input/input_field.dart';
import 'package:frontend/src/components/button.dart';
import 'package:frontend/src/features/login/login_page.dart';
```

File: frontend/lib/src/features/sign_up/sign_up_flow.dart [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    return UserFormData(
+      firstName: firstName ?? this.firstName,
+      middleName: middleName ?? this.middleName,
+      lastName: lastName ?? this.lastName,
+      address: address ?? this.address,
+      birthday: birthday ?? this.birthday,
+      username: username ?? this.username,
+      email: email ?? this.email,
+      password: password ?? this.password,
+      passwordConfirmation: passwordConfirmation ?? this.passwordConfirmation,
```
Comment (rdelrosariojr) [inline frontend/lib/src/features/sign_up/sign_up_flow.dart:56]: please add validation for password and confirm password if they match.

**Review by rdelrosariojr — COMMENTED** [review]
reviewed time: 0.8
comments: 4

File: frontend/lib/src/features/sign_up/sign_up_flow.dart (hunk 8 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/sign_up_flow.dart:56]: Added note that validation should be applied in integration.

File: frontend/lib/src/features/sign_up/sign_up_step_two.dart (hunk 7 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/sign_up_step_two.dart:9]: Adjusted

File: frontend/lib/src/features/sign_up/sign_up_step_one.dart (hunk 6 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/sign_up_step_one.dart:11]: Adjusted

File: frontend/lib/src/features/sign_up/components/breadcrumb.dart (hunk 5 above)
Comment (jescabillas) [reply frontend/lib/src/features/sign_up/components/breadcrumb.dart:2]: Adjusted

**Review by rdelrosariojr — APPROVED** [review]
**APPROVED PR11**

reviewed time: 0.82
comments: 4

**Pull request merged**